### PR TITLE
[inductor] Enable mypy checking for torch/_inductor/codegen/triton.py

### PR DESCRIPTION
**Pull request by masnesral**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #109643
* __->__ #109146

Summary: enably mypy chcking for torch/_inductor/codegen/triton.py and make the minimum number of fixes / ignores to get the linter to pass

Test Plan: `lintrunner -a`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

**Comment by masnesral** [comment]
@eellison , I'd started on this one last week so I decided to try to finish it.

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -2279,7 +2285,8 @@ def end_current_reduction_loop():
                         node not in done
                         and fits_in_main_body(other_node)
                         and not (
-                            current_loop_writes & other_node.recursive_predecessors
```
Comment (masnesral) [inline torch/_inductor/codegen/triton.py:2282]: Soooo, `current_write_loops` doesn't exist and this code path must already be broken. I glanced at the history and made a guess that the name should have changed to `current_loop_reduced_writes`, but I don't know what the code is doing exactly so I may be wrong. I see that @Chillee made some changes to this file recently. @Chillee, maybe you can vet this?

Comment (masnesral) [reply torch/_inductor/codegen/triton.py:2282]: Or maybe @ezyang? I see your name around these parts.

Comment (masnesral) [reply torch/_inductor/codegen/triton.py:2282]: Sorry, scratch that. I guess I see @jansel's name most around this section.

Comment (eellison) [reply torch/_inductor/codegen/triton.py:2282]: I think it's defined above, in `generate_node_schedule(self, nodes, numel, rnumel`  Maybe you need `nonlocal current_write_loops`

Comment (masnesral) [reply torch/_inductor/codegen/triton.py:2282]: @eellison, sorry my comment has the wrong variable name. My _change_ actually replaces `current_loop_writes` with `current_loop_reduced_writes`. I don't find `current_loop_writes` anywhere in this file.

Comment (lezcano) [reply torch/_inductor/codegen/triton.py:2282]: I think this one is on @peterbell10?

Comment (peterbell10) [reply torch/_inductor/codegen/triton.py:2282]: The PR is reverted  already anyway, so a rebase should fix it,

**Review by peterbell10 — APPROVED** [review]

**Pull request closed without merging**